src/autocomplete/dataset.js

```diff
@@ -93,7 +93,7 @@ _.mixin(Dataset.prototype, EventEmitter, {
     var renderArgs = [].slice.call(arguments, 2);
     this.$el.empty();
 
-    hasSuggestions = suggestions && suggestions.length;
+    hasSuggestions = suggestions && suggestions.length > 0;
```

Comment:
I'm not sure what the intention is, but its an array in the angular example here: https://github.com/algolia/autocomplete.js/blob/master/examples/basic_angular.html#L58

Regardless of what type it is, I think the expected behavior when an empty array is used would be that the suggestions is considered empty (which is not the case currently).

I'm not sure this really fixes the caching issue, but it definitely results in what I would think is the expected behavior if an empty array is passed to handleSuggestions, right?